### PR TITLE
change interface

### PR DESCRIPTION
**Pull request by okdshin**
`menoh_variable_profile_table_builder_add_input_profile_dims_2/4` is
replaced with `menoh_variable_profile_table_builder_add_input_profile`

### CODE REVIEW

File: include/menoh/menoh.h [hunk 1]
```diff
@@ -107,8 +107,7 @@ void MENOH_API menoh_delete_model_data(menoh_model_data_handle model_data);
  * This struct configure profiles of variables.
  *
  * See
- *  - menoh_variable_profile_table_builder_add_input_profile_dims_2()
- *  - menoh_variable_profile_table_builder_add_input_profile_dims_4()
+ *  - menoh_variable_profile_table_builder_add_input_profile()
```
Comment (msakai) [inline include/menoh/menoh.h]: This commit removes old APIs but how about keep them as deprecated API for the time being to keep source/binary backward compatibility.

Comment (okdshin) [reply include/menoh/menoh.h]: I see. I'll return old interface.

Comment (okdshin) [reply include/menoh/menoh.h]: done.

Comment (msakai) [reply include/menoh/menoh.h]: Thanks! :+1:

**Comment by msakai** [comment]
This PR contains lots of changes and I'm unaware of the current status as a whole.

I think that adding `menoh_variable_profile_table_builder_add_input_profile` is a simple and useful change.
How about merging this change now, and discussing other changes in different PR?

**Comment by okdshin** [comment]
Yes. I'll split them per functional unit and I'll make smaller PRs. Can I close this PR?

**Pull request closed without merging**